include/openssl/evp_kdf.h

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright (c) 2018, Oracle and/or its affiliates.  All rights reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef HEADER_EVP_KDF_H
+# define HEADER_EVP_KDF_H
+
+# include <openssl/ossl_typ.h>
+# include <openssl/kdferr.h>
+# ifdef __cplusplus
+extern "C" {
+# endif
+
+# define EVP_KDF_PBKDF2 NID_id_pbkdf2
+# define EVP_KDF_SCRYPT NID_id_scrypt
+# define EVP_KDF_TLS1_PRF NID_tls1_prf
+# define EVP_KDF_HKDF   NID_hkdf
+
+EVP_KDF_CTX *EVP_KDF_CTX_new_id(int id);
+void EVP_KDF_CTX_free(EVP_KDF_CTX *ctx);
+
+int EVP_KDF_CTX_ctrl(EVP_KDF_CTX *ctx, int cmd, int p1, void *p2);
+int EVP_KDF_CTX_ctrl_str(EVP_KDF_CTX *ctx, const char *type,
+                         const char *value);
+int EVP_KDF_CTX_ctrl_uint64(EVP_KDF_CTX *ctx, int cmd, uint64_t value);
+
+int EVP_KDF_CTX_str2ctrl(EVP_KDF_CTX *ctx, int cmd, const char *str);
+int EVP_KDF_CTX_hex2ctrl(EVP_KDF_CTX *ctx, int cmd, const char *hex);
+
+int EVP_KDF_CTX_md(EVP_KDF_CTX *ctx, int cmd, const char *md);
+
+int EVP_KDF_derive(EVP_KDF_CTX *ctx, unsigned char *key, size_t *keylen);
+
+
+# define EVP_KDF_ALG_CTRL                       0x1000
+
+# define EVP_KDF_CTRL_PASS                      (EVP_KDF_ALG_CTRL)
+# define EVP_KDF_CTRL_SALT                      (EVP_KDF_ALG_CTRL + 1)
+# define EVP_KDF_CTRL_PBKDF2_ITER               (EVP_KDF_ALG_CTRL + 2)
+# define EVP_KDF_CTRL_PBKDF2_MD                 (EVP_KDF_ALG_CTRL + 3)
+# define EVP_KDF_CTRL_TLS_MD                    (EVP_KDF_ALG_CTRL + 4)
+# define EVP_KDF_CTRL_TLS_SECRET                (EVP_KDF_ALG_CTRL + 5)
+# define EVP_KDF_CTRL_TLS_SEED                  (EVP_KDF_ALG_CTRL + 6)
+# define EVP_KDF_CTRL_HKDF_MD                   (EVP_KDF_ALG_CTRL + 7)
```

Comment:
Why not one generic `EVP_KDF_CTRL_MD`?

Reply:
That makes sense.
I made the change and added a generic `EVP_KDF_CTRL_ITER` as well.